CodeU_Assignement_2/src/codeu_assignement_2/KemBinaryTree.java

```diff
@@ -0,0 +1,394 @@
+package codeu_assignement_2;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Stack;
+import java.util.Queue;
+import java.util.LinkedList;
+
+/**
+ *
+ * @author Karine
+ * 
+ * Binary tree with a tree node
+ * Each node points to parent, and left/right child
+ * 
+ */
+public class KemBinaryTree <E> {
+    /* Data members of Tree */
+    protected KemTreeNode<E> m_root=null;
+    
+    /* */
+    public KemBinaryTree(){} 
+            
+    /*  Check if the list is empty  */
+    public boolean isEmpty() { return (m_root == null); }
+    
+    /* Allow us to decide where to add items */
+    public KemTreeNode<E> getRoot() { return m_root;}
+    
+    /* Q1: Get all ancestors */
+    public void printAncestors(KemTreeNode<E> child)
+    {  
+        /* Check if child isn't null */

```

Comment:
In many cases, it's also fine to leave out comments if it'd obvious what's going on.